### DISCUSSION TITLE
What's your roadmap for PostgreSQL?

### DISCUSSION THREAD

**Discussion by ghost** [moj-analytical-services/splink#1657] [Ideas]
I'm curious what your plans are for fleshing out the Postgres functionality. Could UDFs based on [pg_similarity](https://github.com/eulerto/pg_similarity/tree/master) fill the gap that seems to prevent most of the template and comparison libraries getting a PostgreSQL version?

**Comment by RobinL** [comment]
To be honest, Postgres support  was contributed by an external user and it's not something we use internally, so we're not likely to be able to prioritise any active development on it by the core Splink team.

If you're able to find and register similarity functions, you can use them already by specifying comparisons using the following method:
https://moj-analytical-services.github.io/splink/topic_guides/comparisons/customising_comparisons.html?h=comparison#method-4-providing-the-spec-as-a-dictionary

Since you're specifying them as SQL code, the only constraint is that the SQL code must execute successfully within Postgres.  i.e. so long as the functions exist and are known to Postgres, you can use whatever you want, they don't need to be 'added to Splink' 


We're always happy for contributions if you're interested in adding a specific bit of functionality

**Reply by ghost** [reply]
That's great, thanks Robin!

**Reply by ADBond** [reply]
Also potentially of use is the [Postgres scanner duckdb extension](https://duckdb.org/docs/extensions/postgres_scanner) which allows duckdb to run on tables that live in a postgres database. I don't think anyone on the team has had a proper look at this yet, but this is potentially a way to leverage the functions available in duckdb for use on data that's in postgres
